src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -145,14 +145,34 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
     val spark = df.sparkSession
     val dfColumnNames = df.schema.fieldNames
     val indexedColumns = indexConfig.indexedColumns
-    val includedColumns = indexConfig.includedColumns
+    val includeColumns = indexConfig.includedColumns.include
+    val excludeColumns = indexConfig.includedColumns.exclude
     val resolvedIndexedColumns = ResolverUtils.resolve(spark, indexedColumns, dfColumnNames)
-    val resolvedIncludedColumns = ResolverUtils.resolve(spark, includedColumns, dfColumnNames)
+    val resolvedIncludeColumns = ResolverUtils.resolve(spark, includeColumns, dfColumnNames)
+    val resolvedExcludeColumns = ResolverUtils.resolve(spark, excludeColumns, dfColumnNames)
+
+    (resolvedIndexedColumns, resolvedIncludeColumns, resolvedExcludeColumns) match {
+      case (Some(indexed), Some(include), Some(exclude)) =>
+        if (include.intersect(exclude).nonEmpty) {
+          throw HyperspaceException(
+            "IncludedColumns include and exclude columns can not overlap.")
+        }
+
+        // If IndexConfig has both 'include' and 'exclude' columns, we pick the columns
+        // listed in indexConfig.include as index's included columns and ignore `exclude`.
```

Comment:
I think it would be better to throw an exception if a user set both configs. It might cause some confusion to users.